src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/StaffResourceIntTest.kt

```diff
@@ -0,0 +1,186 @@
+package uk.gov.justice.hmpps.prison.api.resource.impl
+
+import com.microsoft.applicationinsights.TelemetryClient
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.mockito.kotlin.any
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.isNull
+import org.mockito.kotlin.verify
+import org.springframework.boot.test.mock.mockito.SpyBean
+import org.springframework.http.MediaType
+
+class StaffResourceIntTest : ResourceTest() {
+
+  @SpyBean
+  protected lateinit var telemetryClient: TelemetryClient
+
+  @Nested
+  @DisplayName("GET /api/staff/roles/{agencyId}/role/{role}")
+  inner class StaffRoles {
+    @Test
+    fun `should return not found if agency does not exist for user`() {
+      webTestClient.get()
```

Comment:
The majority of tests were copied over from the cucumber tests - with additional ones for the role checks